cot-cli/src/utils.rs

```diff
@@ -67,20 +70,373 @@ impl StatusType {
     }
 }
 
-pub(crate) fn find_cargo_toml(starting_dir: &Path) -> Option<PathBuf> {
-    let mut current_dir = starting_dir;
+#[derive(Debug)]
+pub(crate) struct WorkspaceManager {
+    root_manifest: Manifest,
+    package_manifests: HashMap<String, ManifestEntry>,
+}
+
+#[derive(Debug)]
+struct ManifestEntry {
+    path: PathBuf,
+    manifest: Manifest,
+}
+impl WorkspaceManager {
+    pub(crate) fn from_cargo_toml_path(cargo_toml_path: &Path) -> anyhow::Result<Self> {
+        let cargo_toml_path = cargo_toml_path
+            .canonicalize()
+            .context("unable to canonicalize path")?;
+
+        let manifest =
+            Manifest::from_path(&cargo_toml_path).with_context(|| "unable to read Cargo.toml")?;
```

Comment:
```suggestion
            Manifest::from_path(&cargo_toml_path).context("unable to read Cargo.toml")?;
```

I guess we should be able to change this like so, just like `cot-cli/src/migration_generator.rs` is modified, too!